tests/Fixer/ArrayNotation/MultilineLongArrayFixerTest.php

```diff
@@ -0,0 +1,434 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ArrayNotation;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Philippe Bouttereux <philippe.bouttereux@gmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ArrayNotation\MultilineLongArrayFixer
+ */
+final class MultilineLongArrayFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param array<string, mixed> $config
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null, array $config = []): void
+    {
+        $this->fixer->configure($config);
+
+        $this->doTest($expected, $input);
+    }
+
+    public static function provideFixCases(): iterable
+    {
+        yield 'Old style array' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = array(
+            'foo',
+            'bar' => 'baz'
+            );
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $foo = array('foo','bar' => 'baz');
+            INPUT,
+        ];
+
+        yield 'Old style array with comments' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = array /* comment */ (
+            'foo',
+            'bar' => 'baz'
+            );
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $foo = array /* comment */ ('foo','bar' => 'baz');
+            INPUT,
+        ];
+
+        yield 'Empty array with zero max length' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = [];
+            EXPECTED,
+            null,
+            ['max_length' => 0],
+        ];
+
+        yield 'Empty array with negative max length' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = [];
+            EXPECTED,
+            null,
+            ['max_length' => -1],
+        ];
+
+        yield 'Single-line array' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = [
+            'foo',
+            'bar' => 'baz',
+            ];
+            EXPECTED,
+            <<<'INPUT'
+            <?php
+            $foo = ['foo','bar' => 'baz',];
+            INPUT,
+        ];
+
+        yield 'Single-line array shorter than max_length' => [
+            <<<'EXPECTED'
+            <?php
+            $foo = ['foo','bar' => 'baz',];
+            EXPECTED,
+            null,
+            ['max_length' => 30],
+        ];
+
+        yield 'Single-line array with negative max_length' => [
```

Comment:
There is no need for case testing some code is not changing if there is a test where the same code occurs as expected (like this one in the case `Multi line array with negative max_length`) - expected code is always checked that is no changing.